v2/pkg/extractors/extract.go

```diff
@@ -16,10 +16,11 @@ func (e *Extractor) Extract(resp *http.Response, body, headers string) map[strin
 			return e.extractRegex(headers)
 		} else {
 			matches := e.extractRegex(headers)
-			if len(matches) > 0 {
```

Comment:
This looked like a bug. If you provided a extractor with `all` as the part and your regex matched anything in the headers you would not get the extractor on the body.